### ISSUE TITLE
Connection from any tools fails in workflow editor

### ISSUE THREAD

**Issue by mmiladi** [galaxyproject/galaxy#5163]
In the dev branch I encounter a problem with workflow editor. This issue is not happening with release 17.09. 
While creating a workflow of ~10 tools, after a certain point, connecting output of a specific tool to another (both sides data-collection) prevents creating any other link. The connecting arrow, which suppose to popup by moving the mouse over it, does not show up anymore. Instead the mouse drags & moves the tool rectangle. It feels to be a client side issue, though not sure. 
This is the js error message from the browser console:

```
extended.bundled.js?v=1512735965:49 Uncaught ReferenceError: inputsFilled is not defined
    at I.n._inputFilled (extended.bundled.js?v=1512735965:49)
    at I.n.canAccept (extended.bundled.js?v=1512735965:49)
    at I.n.onDropInit (extended.bundled.js?v=1512735965:49)
    at HTMLDivElement.dispatch (libs.bundled.js:24)
    at HTMLDivElement.e.dispatch (libs.bundled.js:33)
    at HTMLDivElement.e.dispatch (jquery.event.drag.js?v=1512735965:1)
    at HTMLDivElement.<anonymous> (jquery.event.drag.js?v=1512735965:1)
    at Function.each (libs.bundled.js:14)
    at t.fn.init.each (libs.bundled.js:14)
    at Object.hijack (jquery.event.drag.js?v=1512735965:1)
_inputFilled	@	extended.bundled.js?v=1512735965:49
canAccept	@	extended.bundled.js?v=1512735965:49
onDropInit	@	extended.bundled.js?v=1512735965:49
dispatch	@	libs.bundled.js:24
e.dispatch	@	libs.bundled.js:33
e.dispatch	@	jquery.event.drag.js?v=1512735965:1
(anonymous)	@	jquery.event.drag.js?v=1512735965:1
each	@	libs.bundled.js:14
each	@	libs.bundled.js:14
hijack	@	jquery.event.drag.js?v=1512735965:1
handler	@	jquery.event.drop.js?v=1512735965:1
init	@	jquery.event.drag.js?v=1512735965:1
dispatch	@	libs.bundled.js:24
e.dispatch	@	libs.bundled.js:33
e.dispatch	@	jquery.event.drag.js?v=1512735965:1
v.handle	@	libs.bundled.js:24
```

**Comment by dannon** [comment]
I think I see what the problem is here, working on a fix.

**Comment by mmiladi** [comment]
Very good. Thanks for checking it. Some further info: When I import the complete workflow from 17.09 to the dev instance, same problem happens. Removing about 3 of the links (two of them data-collection), allows me again to link tools.

**Comment by dannon** [comment]
Should be fixed in #5166 (https://github.com/galaxyproject/galaxy/pull/5166/commits/5602f911e757e2c08485166e61c39d7c854a7446 specifically, if you want to pull/test that one change).

This actually was broken ~3 years ago, it looks like, we're only now enforcing rules that would have shown the error instead of simply unexpected behavior.

**Comment by mmiladi** [comment]
@dannon thanks a lot. The problem is resolved now.
